### PR TITLE
Remove code from Compiler that is not used anymore

### PR DESCRIPTION
**Pull request by alalazo**
#45784 unified the search mechanism of `spack compiler find` and `spack external find`. This PR removes attributes from `Compiler` classes that were mainly used in the old search mechanism.

**Prerequisites**

- [x] #46037
- [x] #46038

### CODE REVIEW

File: var/spack/repos/builtin/packages/boost/package.py
Comment (hainest) [inline var/spack/repos/builtin/packages/boost/package.py]: I'm not sure if Boost builds with the oneapi compiler. I'll have to double-check.

**Review by hainest — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/boost/package.py
Comment (hainest) [inline var/spack/repos/builtin/packages/boost/package.py]: 1. Typo in 'satisfies'
2. Should this just be `if self.spec.satisfies(cc)`?

**Review by hainest — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/boost/package.py
Comment (alalazo) [reply var/spack/repos/builtin/packages/boost/package.py]: Thanks for catching this typo. I was planning to extract package changes from this PR and ping people there.

File: lib/spack/spack/test/architecture.py
Comment (alalazo) [inline lib/spack/spack/test/architecture.py:133]: One of two tests I had to disable on Windows

File: lib/spack/spack/test/compilers/basics.py
Comment (alalazo) [inline lib/spack/spack/test/compilers/basics.py:592]: The other one

**Review by alalazo — COMMENTED** [review]
I had to disable 2 tests on Windows, because the logic for determining if a toolchain is "mixed" changes here. 

I don't think Windows is affected in any way[^1], but I'll let @johnwparent comment on that.

[^1]: The `is_mixed_toolchain` function is used to inject compiler flags via the wrapper, which Windows is not using afaik

**Review by haampie — APPROVED** [review]

**Comment by johnwparent** [comment]
Already merged, but FWIW @alalazo this is perfectly fine from the Windows side. All the flag injection logic will require a refactor anyway once the Windows compiler wrapper lands, so perfectly fine to skip/remove whatever you need from Windows for the time being. Appreciate the ping.

**Comment by alalazo** [comment]
Now that pipelines are back, this seems to be causing failures on aws-pcluster setup. I'll investigate asap.

**Comment by alalazo** [comment]
Fix in https://github.com/spack/spack/pull/46505

**Pull request merged**
